code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -558,24 +560,24 @@
 			module.integrated_ai << "[message]"
 			. = 1
 
-/obj/item/weapon/rig/equipped(mob/living/carbon/human/M)
+//Delayed equipping of rigs
+/obj/item/weapon/rig/pre_equip(var/mob/user, var/slot)
+	if (slot == rig_wear_slot)
+		if(seal_delay > 0)
+			user.visible_message("<font color='blue'>[user] starts putting on \the [src]...</font>", "<font color='blue'>You start putting on \the [src]...</font>")
+			if(!do_after(user,seal_delay,src))
+				return 1 //A nonzero return value will cause the equipping operation to fail
+
+
+/obj/item/weapon/rig/equipped(var/mob/user, var/slot)
```

Comment:
Equipped is called AFTER an item is equipped, so it was a fundamentally wrong place to have the doafter for putting it on